packages/pages-shared/metadata-generator/parseRedirects.ts

```diff
@@ -109,6 +109,19 @@ export function parseRedirects(input: string): ParsedRedirects {
 			continue;
 		}
 
+		// We want to always block the `/* /index.html` redirect - this will cause TOO_MANY_REDIRECTS errors as
+		// the asset server will redirect it back to `/`, removing the `/index.html`. This is the case for regular
+		// redirects, as well as proxied (200) rewrites. We only want to run this on relative urls
+		if (/\/\*?$/.test(from) && /\/index(.html)?$/.test(to) && !urlHasHost(to)) {
```

Comment:
This will fire on `/* /foo/index.html`, can we add some tests for this too? Also a test for the has host condition. 

Then just some ok conditions e.g `/* /foo` should be ok and not be invalid.